### PR TITLE
DDOS-141 - fixed perpetual retry

### PR DESCRIPTION
**Pull request by yakovshkolnikov**

### CODE REVIEW

File: AlamofireSessionRenewer/Classes/AlamofireSessionRenewer.swift [hunk 1]
```diff
@@ -70,7 +70,7 @@ open class AlamofireSessionRenewer: RequestRetrier {
     // MARK: - RequestRetrier protocol implementation
     
     public func should(_ manager: SessionManager, retry request: Request, with error: Error, completion: @escaping RequestRetryCompletion) {
-        if (error as NSError).domain == errorDomain, (error as NSError).code == authenticationErrorCode {
+        if credential != nil, (error as NSError).domain == errorDomain, (error as NSError).code == authenticationErrorCode {
```
Comment (DDKirillUshkov) [inline AlamofireSessionRenewer/Classes/AlamofireSessionRenewer.swift]: @yakovshkolnikov May we create open method in AlamofireSessionRenewer which validates credential so clients may provide their own way to validate credential?

**Review by DDKirillUshkov — CHANGES_REQUESTED** [review]

File: AlamofireSessionRenewer/Classes/AlamofireSessionRenewer.swift (hunk 1 above)
Comment (yakovshkolnikov) [reply AlamofireSessionRenewer/Classes/AlamofireSessionRenewer.swift]: We already have: open func isCredentialEqual(to request: Request) -> Bool

Comment (DDKirillUshkov) [reply AlamofireSessionRenewer/Classes/AlamofireSessionRenewer.swift]: So why are we using credential != nil check here?

Comment (DDKirillUshkov) [reply AlamofireSessionRenewer/Classes/AlamofireSessionRenewer.swift]: Lets create separate method isCredentialEmpty so logic in should retry method will be more readable

**Review by DDKirillUshkov — APPROVED** [review]

**Pull request merged**